clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/security/KubernetesCredentialFactory.java

```diff
@@ -43,6 +43,10 @@ default void validateAccount(KubernetesConfigurationProperties.ManagedAccount ma
   default String getKubeconfigFile(
       ConfigFileService configFileService,
       KubernetesConfigurationProperties.ManagedAccount managedAccount) {
+    if (managedAccount.isServiceAccount()) {
```

Comment:
I'd like to avoid more uses of the `isServiceAccount` flag because I think it's redundant to other config settings.  The only other place this flag is currently used is [here](https://github.com/spinnaker/clouddriver/blob/658bac1b8b21121cdb1ac6fbfcb76cec16d5a410/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/job/KubectlJobExecutor.java#L530) where it's gating a block of logic...but each item in the block is itself gated by its own config so I've been meaning to see if I can remove that outer gate.

In other words, I think that `isServiceAccount` should not exist as a separate flag, but should be equivalent to just having both `kubeconfigFile` and `kubeconfigContents` be empty.

In this case, could we omit returning `System.getProperty("user.home") + "/.kube/config";` if both of those are empty?  I believe that's one of the default places that `kubectl` will search if you don't explicitly supply a path, so we should not need to supply it explicitly.

Reply:
Sounds right, I see the `--kubeconfig` parameter is only appended to `kubectl` if kubeconfig is not empty, and if it's not appended it will just use this default location